windows/shared/logging/logging-plog-windows/include/logging/windows/PlogDebugWrapper.hxx

```diff
@@ -0,0 +1,45 @@
+
+#ifndef PLOG_DEBUG_WRAPPER_HXX
+#define PLOG_DEBUG_WRAPPER_HXX
+
+#include <windows.h>
+
+#include <plog/Appenders/DebugOutputAppender.h>
+#include <plog/Appenders/DynamicAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+namespace logging
+{
+namespace plog
+{
+
+template <typename Formatter>
+class DebugWrapperAppender : public ::plog::DynamicAppender
+{
+public:
+    /**
+     * @brief Construct a new DebugWrapperAppender object
+     * This class adds a DebugOutputApender iff the DEBUG macro is set and IsDebuggerPresent() is true
+     *  otherwise this class just does the same as the input Appender
+     * @param providerHandle
+     */
+    DebugWrapperAppender(::plog::IAppender *appender)
+    {
+        ::plog::DynamicAppender::addAppender(appender);
+#ifdef DEBUG
+        if (IsDebuggerPresent()) {
+            ::plog::DynamicAppender::addAppender(&m_debugAppender);
+        }
+#endif
```

Comment:
Note that a debugger can be attached to a non-debug build where `DEBUG` isn't defined. Given that, it doesn't seem this wrapper is necessary; we can just add a `::plog::DebugOutputAppender` to the chain of appenders all the time (on Windows). I looked at its implementation and all it does is call `OutputDebugStringW` which is a no-op when no debugger is attached. So, we don't even need to check `IsDebuggerPresent()` like I thought.

Reply:
I decided that it would be better to have this simple wrapper rather than repeat the same 4 lines of constructing the DebugAppender, the DynamicAppender, and adding the DebugAppender and the normal appender to the DynamicAppender. It would also make it easier in the future to extend the wrapper to linux if we wanted to.